mlir/include/Driver/CatalystLLVMTarget.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "mlir/IR/DialectRegistry.h"
+#include "mlir/Support/LogicalResult.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/IR/Module.h"
+
+#include "CompilerDriver.h"
+
+namespace driver {
```

Comment:
I am not sure that makes sense. Right now the `catalyst` namespace contains the catalyst dialect which is different from the driver. I guess you could make the argument of having `catalyst::driver` and `catalyst::dialect` as distinct, but then we would also have `quantum::dialect` and `gradient::dialect` to keep consistent. If you insist, I will change, but overall I think the driver is distinct from the dialect and therefore should remain in different namespaces.

Reply:
Hmm that's a good point. I think the dialect should live in `catalyst::catalyst` then. All our utility functions live at least in the `::catalyst` namespace, a top-level `::driver` namespace is odd since we are also dealing with other projects and the first level namespace has always delineated the project so far (mlir, llvm, catalyst, ..).

Reply:
I moved these ones to `catalyst::driver` but I have not changed `catalyst::catalyst`. That can be a refactoring done later (which I can also do) but let's keep this PR only to the compiler driver for now.